### DISCUSSION TITLE
okd installation errors

### DISCUSSION THREAD

**Discussion by pawel-krol** [okd-project/okd#1083] [Q&A]
hello, i trying to install okd using openshift-install script on vspherean  i got a error like this:

INFO Waiting up to 30m0s for bootstrapping to complete...
ERROR Cluster operator authentication Degraded is True with APIServerDeployment_UnavailablePod::IngressStateEndpoints_MissingSubsets::OAuthServerConfigObservation_Error::OAuthServerServiceEndpointAccessibleController_SyncError::OAuthServerServiceEndpointsEndpointAccessibleController_SyncError::RouterCerts_NoRouterCertSecret: APIServerDeploymentDegraded: 2 of 2 requested instances are unavailable for apiserver.openshift-oauth-apiserver (crashlooping container is waiting in apiserver-7d496fd7df-b8mkj pod, crashlooping container is waiting in apiserver-7d496fd7df-rn4wj pod)
ERROR IngressStateEndpointsDegraded: No subsets found for the endpoints of oauth-server
ERROR OAuthServerConfigObservationDegraded: secret "v4-0-config-system-router-certs" not found
ERROR OAuthServerServiceEndpointAccessibleControllerDegraded: Get "https://172.30.186.230:443/healthz": dial tcp 172.30.186.230:443: connect: connection refused
ERROR OAuthServerServiceEndpointsEndpointAccessibleControllerDegraded: oauth service endpoints are not ready
ERROR RouterCertsDegraded: neither the custom secret/v4-0-config-system-router-certs -n openshift-authentication or default secret/oauth-openshift -n openshift-authentication could be retrieved: secret "v4-0-config-system-router-certs" not found
INFO Cluster operator authentication Available is False with APIServerDeployment_NoPod::APIServices_PreconditionNotReady::OAuthServerServiceEndpointAccessibleController_EndpointUnavailable::OAuthServerServiceEndpointsEndpointAccessibleController_ResourceNotFound::ReadyIngressNodes_NoReadyIngressNodes: APIServerDeploymentAvailable: no apiserver.openshift-oauth-apiserver pods available on any node.
INFO APIServicesAvailable: PreconditionNotReady
INFO OAuthServerServiceEndpointAccessibleControllerAvailable: Get "https://172.30.186.230:443/healthz": dial tcp 172.30.186.230:443: connect: connection refused
INFO OAuthServerServiceEndpointsEndpointAccessibleControllerAvailable: endpoints "oauth-openshift" not found
INFO ReadyIngressNodesAvailable: Authentication requires functional ingress which requires at least one schedulable and ready node. Got 0 worker nodes, 2 master nodes, 0 custom target nodes (none are schedulable or ready for ingress pods).
INFO Cluster operator baremetal Disabled is True with UnsupportedPlatform: Nothing to do on this Platform
ERROR Cluster operator etcd Degraded is True with EnvVarController_Error::InstallerController_Error::RevisionController_ContentCreationError::ScriptController_Error: EnvVarControllerDegraded: can't update etcd pod configurations because scaling is currently unsafe: 3 nodes are required, but only 2 are available
ERROR InstallerControllerDegraded: missing required resources: [configmaps: etcd-scripts,restore-etcd-pod, configmaps: etcd-metrics-proxy-client-ca-1,etcd-metrics-proxy-serving-ca-1,etcd-peer-client-ca-1,etcd-pod-1,etcd-serving-ca-1, secrets: etcd-all-certs-1]
ERROR RevisionControllerDegraded: configmaps "etcd-pod" not found
ERROR ScriptControllerDegraded: "configmap/etcd-pod": missing env var values
INFO Cluster operator etcd Progressing is True with NodeInstaller: NodeInstallerProgressing: 2 nodes are at revision 0; 0 nodes have achieved new revision 1
INFO Cluster operator etcd Available is False with StaticPods_ZeroNodesActive: StaticPodsAvailable: 0 nodes are active; 2 nodes are at revision 0; 0 nodes have achieved new revision 1
INFO Cluster operator etcd RecentBackup is Unknown with ControllerStarted:
INFO Cluster operator ingress Available is Unknown with IngressDoesNotHaveAvailableCondition: The "default" ingress controller is not reporting an Available status condition.
INFO Cluster operator ingress Progressing is True with Reconciling: Not all ingress controllers are available.
ERROR Cluster operator ingress Degraded is Unknown with IngressDoesNotHaveDegradedCondition: The "default" ingress controller is not reporting a Degraded status condition.
INFO Cluster operator insights Disabled is False with AsExpected:
ERROR Cluster operator kube-apiserver Degraded is True with StaticPods_Error: StaticPodsDegraded: pod/kube-apiserver-test-htcpk-master-0 container "kube-apiserver" is waiting: CrashLoopBackOff: back-off 2m40s restarting failed container=kube-apiserver pod=kube-apiserver-test-htcpk-master-0_openshift-kube-apiserver(0413b00b-6fca-475c-b6c4-ab812beb8168)
ERROR StaticPodsDegraded: pod/kube-apiserver-test-htcpk-master-0 container "kube-apiserver-check-endpoints" is waiting: CrashLoopBackOff: back-off 2m40s restarting failed container=kube-apiserver-check-endpoints pod=kube-apiserver-test-htcpk-master-0_openshift-kube-apiserver(0413b00b-6fca-475c-b6c4-ab812beb8168)
INFO Cluster operator kube-apiserver Progressing is True with NodeInstaller: NodeInstallerProgressing: 2 nodes are at revision 0; 0 nodes have achieved new revision 3
INFO Cluster operator kube-apiserver Available is False with StaticPods_ZeroNodesActive: StaticPodsAvailable: 0 nodes are active; 2 nodes are at revision 0; 0 nodes have achieved new revision 3
INFO Cluster operator monitoring Progressing is True with RollOutInProgress: Rolling out the stack.
ERROR Cluster operator monitoring Degraded is Unknown with :
INFO Cluster operator monitoring Available is Unknown with :
INFO Cluster operator network ManagementStateDegraded is False with :
INFO Cluster operator network Progressing is True with Deploying: Deployment "openshift-network-diagnostics/network-check-source" is waiting for other operators to become ready
ERROR Cluster operator openshift-apiserver Degraded is True with APIServerDeployment_UnavailablePod: APIServerDeploymentDegraded: 2 of 2 requested instances are unavailable for apiserver.openshift-apiserver (crashlooping container is waiting in apiserver-58b6587f56-hdqbs pod, crashlooping container is waiting in apiserver-58b6587f56-m4nsb pod)
INFO Cluster operator openshift-apiserver Available is False with APIServerDeployment_NoPod::APIServices_PreconditionNotReady: APIServerDeploymentAvailable: no apiserver.openshift-apiserver pods available on any node.
INFO APIServicesAvailable: PreconditionNotReady
INFO Cluster operator operator-lifecycle-manager-packageserver Available is False with ClusterServiceVersionNotSucceeded: ClusterServiceVersion openshift-operator-lifecycle-manager/packageserver observed in phase Failed with reason: InstallCheckFailed, message: install timeout
INFO Cluster operator operator-lifecycle-manager-packageserver Progressing is True with : Working toward 0.18.3
ERROR failed to lookup master.1 ipv4 address: Post "https://172.70.10.120/sdk": context deadline exceeded
INFO Pulling debug logs from the bootstrap machine
INFO Bootstrap gather logs captured here "/opt/okd/log-bundle-20220128132017.tar.gz"
ERROR Bootstrap failed to complete: timed out waiting for the condition
ERROR Failed to wait for bootstrapping to complete. This error usually happens when there is a problem with control plane hosts that prevents the control plane operators from creating the control plane.
FATAL Bootstrap failed to complete

Any suggestion ?

**Comment by vrutkovs** [comment]
Please attach [log bundle](https://docs.okd.io/latest/installing/installing-troubleshooting.html#installation-bootstrap-gather_installing-troubleshooting)

**Reply by pawel-krol** [reply]
Attached

**Comment by dkoci** [comment]
First check if you use proper fcos version. As for my experience, this is the main issue in almost all my cases :-)

**Reply by pawel-krol** [reply]
I have no idea, installation script is downloading it. How to check it ?

**Comment by pawel-krol** [comment]
[log-bundle-20220128132017.tar.gz](https://github.com/openshift/okd/files/7958858/log-bundle-20220128132017.tar.gz)

**Reply by vrutkovs** [reply]
bootstrap instance hostname is `localhost`, your DHCP is misconfigured.
2 masters out of 3 joined the cluster, no logs from the third one:
>EnvVarControllerDegraded: can't update etcd pod configurations because scaling is currently unsafe: 3 nodes are required, but only 2 are available

**Reply by pawel-krol** [reply]
Thx, DHCP i will check but boostrap hostname  is set by installation script (openshift-install)

**Reply by pawel-krol** [reply]
What is wrong with dhcp ? i checked and i'm not able to find problem

**Reply by pawel-krol** [reply]
Strange situation, after few hours and restarts c luster start but for example without ingress and some other services.
